Add support for to_javascripts url property

richtera opened the pull request:
1. Unprotect make_url.
2. Add simple delegator for model.url.

This will allow

``` mustache

---
custom: some.js

---
{{# page.custom?to_javascripts }}
   <script type="text/javascript" src="{{ url }}"></script>
{{/ page.custom?to_javascripts }}
```

to render out

``` html
<script type="text/javascript" src="/assets/javascripts/some-...hash....js"></script>
```

using the correct fingerprinted javascript file url.

richtera committed:
Add support for to_javascripts url property

1. Unprotect make_url.
2. Add simple delegator for model.url.

richtera commented:
NOTE: I has problems using the master branch of ruhoh so this was originally written against the 2.5 branch. Is there a problem with the master branch or is it something on my machine?

plusjade commented:
> NOTE: I has problems using the master branch of ruhoh so this was originally written against the 2.5 branch. Is there a problem with the master branch or is it something on my machine?

Well that depends on what the problem was =p. Did the tests fail (known issue) or were you not able to get it running at all?

richtera commented:
Whenever I try to ruhoh compile I get some kind of method error. Let me set it up again (I had rolled back to 2.5 and it worked fine so I just left it there.)
Thanks
Andy

On Dec 9, 2013, at 8:37 PM, Jade Dominguez notifications@github.com wrote:

> NOTE: I has problems using the master branch of ruhoh so this was originally written against the 2.5 branch. Is there a problem with the master branch or is it something on my machine?
> 
> Well that depends on what the problem was =p. Did the tests fail (known issue) or were you not able to get it running at all?
> 
> —
> Reply to this email directly or view it on GitHub.

richtera committed:
Also add support for javascripts.all

Add id, path and url as property of javascript object to allow mappings
to be rendered as well.

richtera commented:
The latest push also supports things like

``` mustache
<script type="text/javascript">
  function mapUrl(id) {
    switch (id) {
{{# javascripts.all }}
    case '{{id}}': return '{{url}}';
{{/ javascripts.all }}
    }
    return null;
  }
</script>
```

Which is a function which could be used within something like require.js to load the correct URL according to the required url.

richtera committed:
Missing change.

richtera committed:
Add tests for javascript url features.

richtera committed:
Typo in scenario

Should be
'And this file should have the fingerprinted javascripts "..."'
instead of
'And the file should have the fingeprinted javascript "..."'

plusjade commented:
@richtera this is very cool. Thanks for taking the time to do this! :smiley:
